### PR TITLE
Unicode render fixes

### PR DESCRIPTION
**Pull request by hackergrrl**
Uses the module `wcwidth` throughout cabal-cli's render logic.

This modifies several key drawing routines to use the wcwidth module, in order to compute the visual width of Unicode code points on a fixed-width display (i.e. a terminal). A great deal of render bugs apparent on this client are fixed by this.

Depends on:
- [x] https://github.com/mafintosh/ansi-diff/pull/8

### CODE REVIEW

**Comment by hackergrrl** [comment]
Also! For reasons I have not investigated, rendering seems to be significantly faster! 😎

**Comment by hackergrrl** [comment]
> i was going to ask why markdown-shim.js was deleted since that seemed unrelated to the patch

Yeah! I found cases in the Starlight Union channel where it breaks rendering on my terminal emulator. `supports-hyperlinks` claims my terminal supports ANSI links (didn't know that was a thing!), but it in fact does not seem to. I didn't dig deep into the markdown shim piece *at all* just to be clear, it just seemed like a low-impact thing to remove for now, since the only markdown thing it did was replacing `[foo](https://bar.com)` patterns.

**Comment by hackergrrl** [comment]
Thanks @cblgh, I appreciate you taking the time!

File: views.js [hunk 1]
```diff
@@ -245,16 +245,21 @@ function renderMessages (state, width, height) {
...
+      // TODO(kira): These don't wrap yet & ought to!
+      accum.push(' * ' + msg.formatted)
+      return accum
     }
-    if (state.config.messageIndent === 'nick') {
-      indent += nickLength + 3 // + space and <>
-    }
-    accum.push.apply(accum, util.wrapAnsi(msg.formatted, width, indent))
+
+    const indent = util.strwidth(msg.timestamp + msg.author + msg.emote) + 1
```
Comment (cblgh) [inline views.js:255]: i didn't follow what the `+1` was, could you add a comment that explains what it is?

**Review by cblgh — APPROVED** [review]
lgtm! small comment on a magic number, in case the rendering logic changes in the future and we're scratching our heads over what it means

i mostly checked out the code and ran it a bit to see what the client was like with this change :]

File: views.js (hunk 1 above)
Comment (hackergrrl) [reply views.js]: It's the space between the `09:13pm <foo>` author prefix and the actual message content. I've revised this a bit so there's no `+ 1`.

**Comment by hackergrrl** [comment]
Thank you for looking this over, @cblgh.

**Comment by hackergrrl** [comment]
`15.0.1`

**Comment by hackergrrl** [comment]
`15.0.2`. I had `package-lock.json` files disabled on my npm setup, and so there's another patch release that includes that file being updated, which I *believe* is what `npm install` uses.

**Pull request merged**